### ISSUE TITLE
Security: DoS

### ISSUE THREAD

**Issue by rjacoby** [christian-bromann/rgb2hex#6]
`npm audit` reports 'Moderate' DoS issue.

See [https://nodesecurity.io/advisories/647](https://nodesecurity.io/advisories/647 )

```
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ Moderate      │ Denial of Service                                            │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ rgb2hex                                                      │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ No patch available                                           │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ webdriverio [dev]                                            │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ webdriverio > rgb2hex                                        │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://nodesecurity.io/advisories/647                       │
└───────────────┴──────────────────────────────────────────────────────────────┘
```

**Comment by christian-bromann** [comment]
Is this still an issue with the latest version? We tried to fix the regex string.

**Comment by alexjfno1** [comment]
I have the same warning using the latest `0.1.4`.

**Comment by rjacoby** [comment]
Yes, `0.1.4`.

```
└─┬ webdriverio@4.13.1
  └── rgb2hex@0.1.4
```

**Comment by radiovisual** [comment]
~This problem seems to be gone now with `0.1.5`:~

![image](https://user-images.githubusercontent.com/5614571/42305112-bde4f592-8029-11e8-909b-fa3d10a27e6f.png)

~Maybe @rjacoby can confirm?~

Nevermind, the problem persists in `0.1.5`:

![image](https://user-images.githubusercontent.com/5614571/42310224-23b3a8b2-803b-11e8-9afe-1e71a8d1010a.png)

**Comment by radiovisual** [comment]
The RegExp in rgb2hex that is being flagged is here: https://github.com/christian-bromann/rgb2hex/blob/v0.1.0/index.js#L25

**Comment by christian-bromann** [comment]
I modified the regex to make it more strict. This should fix the vulnerability. The one that is reported here: https://nodesecurity.io/advisories/647 is already fixed. Any idea how to revalidate?

**Comment by radiovisual** [comment]
AFAIK, you can't run an audit on local files (only on your local-installed dependencies), which makes it tricky to refactor and validate that the changes you have made are enough to pass the security checks, so I opened this: https://github.com/npm/npm/issues/21155

**Comment by christian-bromann** [comment]
@radiovisual I installed the new released version in a temporary folder to run the audit but apparently it only relies on the [Nodesecurity database](https://nodesecurity.io/advisories/647). The audit doesn't do actual security checks. I opened an issue and ask to update the report but apparently that can take a while. I've seen projects where reports were outdated for more than 3 years. It seems that there is not way to ask the initial auditor to review the project again.
